lib/context.js

```diff
@@ -52,42 +60,65 @@ module.exports = async function getContext() {
 
       const actor = response.data.display_name || 'Unknown User';
 
-      return {
-        ref: process.env.BITBUCKET_TAG
-          ? `refs/tags/${process.env.BITBUCKET_TAG}`
-          : `refs/heads/${process.env.BITBUCKET_BRANCH}`,
-        sha: process.env.BITBUCKET_COMMIT,
-        actor,
-        runId: parseInt(process.env.BITBUCKET_BUILD_NUMBER, 10),
-        payload: {
-          repository: {
-            name: process.env.BITBUCKET_REPO_SLUG,
-            organization: process.env.BITBUCKET_WORKSPACE,
-          },
-          commits: [
-            {
-              message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
-                env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                encoding: 'utf-8',
-              }).trim(),
-              // Get the unix timestamp of the commit then convert it to
-              // an ISO8601 UTC string
-              //
-              // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
-              // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
-              timestamp: new Date(
-                execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
-                  env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                  encoding: 'utf-8',
-                }).trim() * 1000
-              ).toISOString(),
-              author: {
-                name: actor,
+      // BITBUCKET_PR_ID only set on pull request triggered builds
+      // this is how we determine if it's a `push` or `pull_request`
+      const eventName = process.env.BITBUCKET_PR_ID ? 'pull_request' : 'push';
+
+      switch (eventName) {
+        case 'push':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              repository: {
+                name: process.env.BITBUCKET_REPO_SLUG,
+                // TODO we should remove this in a future release
+                organization: process.env.BITBUCKET_WORKSPACE,
+              },
+              organization: {
+                login: process.env.BITBUCKET_WORKSPACE,
               },
+              commits: [
+                {
+                  message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
```

Comment:
this will return the commit message for only the head commit. what if there are multiple commits? should we not 

from the [Atlassian forums](https://community.atlassian.com/t5/Bitbucket-questions/How-do-I-get-the-commit-SHA-merged-in-in-Pipelines-pull-request/qaq-p/2052426#M82820)

> to get the commit hash that is shown in the merge command at Build Setup stage you could use `git log` as below:
> 
> ```sh
> git log -n 1 --pretty=format:"%H" $BITBUCKET_PR_DESTINATION_BRANCH
> ```
> 
> This will print the tip of the destination branch of the pull request, which should match the commit hash presented in Build Setup.
>
> As for getting the files that were changed in the source branch of the pull request, I think you could make use of the git three-way diff, as below : 
>
> ```sh
> git diff $BITBUCKET_PR_DESTINATION_BRANCH...HEAD
> ```
>
> This will list all the files that were changed in the source branch of the Pull Request starting at the last common commit between the source and destination branch of the pull request.

Reply:
[saw this](https://github.com/markosamuli/commitlint-bitbucket-cli/blob/d64395abd105f992568355ef2c0399889e5f5f2c/src/cli.js#L21-L47) in the wild - not sure if it's useful

Reply:
Right now we're only storing the head commit in the changelog, even if they're pushing more than one at a time! For better or worse, that's how the system works for the moment. Agree though it would probably be better in future to copy over the whole git history